softserve/views.py

```diff
@@ -80,35 +70,48 @@ def dashboard():
 
 
 @app.route('/create_node', methods=['GET', 'POST'])
-#@organization_access_required('gluster')
+@organization_access_required('gluster')
 def get_node_data():
-    count = db.session.query(func.count(Vm.id)) \
-            .filter_by(state = 'ACTIVE').scalar()
-    if count >=5:
-        flash('Limit got over!')
-        return render_template('home.html')
+    if request.method == "POST":
+        counts = request.form['counts']
+        name = request.form['node_name']
+        hours_ = request.form['hours']
+        pubkey_ = request.form['pubkey']
+
+        '''Validating the SSH public key'''
+        ssh = sshpubkeys.SSHKey(pubkey_, strict=True)
+        try:
+            ssh.parse()
+        except Exception:
+            return "Invalid SSH key", 400
+
+        '''Validating the machine label'''
+        n = NodeRequest.query.filter_by(node_name=name).first()
+        if n is None:
+            node_request = NodeRequest(
+                user_id=g.user.id,
+                node_name=name,
+                node_counts=counts,
+                hours=hours_,
+                pubkey=pubkey_)
+            db.session.add(node_request)
+            db.session.commit()
+            create_node.apply_async((counts, name, node_request.id, pubkey_),
+                                    seriaizer='json')
+            return redirect('/dashboard')
+        else:
+            flash('Machine label already exists. \
+                   Please choose different name.')
     else:
-        if request.method == "POST":
-            counts = request.form['counts']
-            name = request.form['node_name']
-            hours_ = request.form['hours']
-            pubkey_ = request.form['pubkey']
-            n = NodeRequest.query.filter_by(node_name=name).first()
-            if n is None:
-                node_request = NodeRequest(
-                    user_id=g.user.id,
-                    node_name=name,
-                    node_counts=counts,
-                    hours=hours_,
-                    pubkey=pubkey_)
-                db.session.add(node_request)
-                db.session.commit()
-                create_node(counts, name, node_request, pubkey_)
-                flash('Your VM has been created. Go back to Dashboard')
-            else:
-                flash('Machine label already exists. \
-                       Please choose different name.')
-        return render_template('home.html')
+        count = db.session.query(func.count(Vm.id)) \
+                .filter_by(state='ACTIVE').scalar()
+        if count >= 5:
+            flash('Oops!Limit got over. Try again later')
```

Comment:
This sounds a little tacky. Perhaps "All our available machines are in use. Please wait until we have a slot available"?